File: crates/parser/fixtures/python/sample.py
from db.repository import UserRepository

TIMEOUT = 30


class Service:
    pass


class AuthService(Service):
    def __init__(self, repository: UserRepository) -> None:
        self.repository = repository

    def login(self, user: str) -> bool:
        return self.repository.find_user(user) is not None


def bootstrap(repository: UserRepository) -> AuthService:
    service = AuthService(repository)
    service.login("ada")
    return service

File: crates/parser/fixtures/rust/sample.rs
use crate::db::UserRepository;

pub struct AuthService {
    repo: UserRepository,
}

pub trait Repository {
    fn find_user(&self, name: &str) -> Option<String>;
}

impl AuthService {
    pub fn login(&self, user: &str) -> bool {
        self.repo.find_user(user).is_some()
    }
}

impl Repository for AuthService {
    fn find_user(&self, name: &str) -> Option<String> {
        Some(name.to_string())
    }
}

pub fn bootstrap() {
    let service = AuthService {
        repo: UserRepository,
    };
    service.login("ada");
}
